src/scripts/data_file_processing/healthsystem/equipment/equipment_availability_estimation.py

```diff
@@ -0,0 +1,273 @@
+    final_equipment_availability_df = pd.concat([final_equipment_availability_df, duplicated_district_rows], ignore_index=True)
+
+# For the districts above, drop level 3 for districts because this lies within the city, drop level 4 from Zomba
+final_equipment_availability_df = final_equipment_availability_df.reset_index()
+for district, city in district_mapping.items():
+    final_equipment_availability_df = final_equipment_availability_df.drop(final_equipment_availability_df[(final_equipment_availability_df['District'] == district) & (final_equipment_availability_df['Facility_level'] == '3')].index)
+final_equipment_availability_df = final_equipment_availability_df.drop(final_equipment_availability_df[(final_equipment_availability_df['District'] == 'Zomba') & (final_equipment_availability_df['Facility_level'] == '4')].index)
+
+# Combine Mzimba North and Mzimba South
+final_equipment_availability_df.loc[(final_equipment_availability_df.District == 'Mzimba North') | (final_equipment_availability_df.District == 'Mzimba South'),'District'] = 'Mzimba'
+equipment_data_district_list = final_equipment_availability_df.District.unique()
+model_district_list = pd.read_csv(resourcefilepath / 'demography/ResourceFile_PopulationSize_2018Census.csv')['District'].unique() # Check that the new list of district matches with the model
+assert(set(equipment_data_district_list) == set(model_district_list))
+
+# Collapse data by facility level and district
+final_equipment_availability = final_equipment_availability_df.groupby(['Facility_level', 'District','Item_code', 'Equipment_name' ])[['available', 'functional']].mean().reset_index()
+
+# -------------------------------------------------------------------------------------------------------------------
+# -------------------------------------------------------------------------------------------------------------------
+# Final formatting for compatability with the `Equipment` class
+
+final_equipment_availability_export = final_equipment_availability.copy()
+
+# - Drop any rows with missing data
+final_equipment_availability_export = final_equipment_availability_export.dropna()
+
+# - 'Item_Code' is the name expected in the Equipment class, and this should of type `int`
+final_equipment_availability_export['Item_Code'] = final_equipment_availability_export['Item_code'].astype(int)
+
+# Rename 'facility_level'
+final_equipment_availability_export = final_equipment_availability_export.rename(columns={'Facility_level': 'Facility_Level'})
+
+# Check that every row specifies a facility_level
+assert final_equipment_availability_export['Facility_Level'].isin([
+    '0', '1a', '1b', '2', '3', '4', '5'
+]).all()
+assert final_equipment_availability_export['Facility_Level'].notnull().all()
+
+# - 'Pr_Available' is the name expected in the Equipment class, and is taken to be the probability that an item of equipment is available for use
+#   which is taken to be equivalent to it to being "functional"
```

Comment:
I think we can use the 'functional' column as the final probability of piece of equipment actually being "available" _and_ "usable", right @sakshimohan?

Reply:
Yes, that makes sense.